presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/optimization/JdbcComputePushdown.java

```diff
@@ -132,32 +137,74 @@ public PlanNode visitFilter(FilterNode node, Void context)
                     jdbcFilterToSqlTranslator,
                     oldTableScanNode.getAssignments());
 
-            // TODO if jdbcExpression is not present, walk through translated subtree to find out which parts can be pushed down
-            if (!oldTableHandle.getLayout().isPresent() || !jdbcExpression.getTranslated().isPresent()) {
+            if (!oldTableHandle.getLayout().isPresent()) {
                 return node;
             }
 
+            Optional<JdbcExpression> translated = jdbcExpression.getTranslated();
+            // All filter can pushdown
+            if (translated.isPresent()) {
+                return createNewTableScanNode(oldTableScanNode, oldTableHandle, oldConnectorTable, translated);
+            }
+            else {
+                // Find out which parts can be pushed down
+                List<RowExpression> pushable = new ArrayList<>();
+                List<RowExpression> nonPushable = new ArrayList<>();
+                List<JdbcExpression> translatedExpressions = new ArrayList<>();
```

Comment:
`pushable` do not have to co-exist; we only use `pushable.isEmpty`; that holds true for `translatedExpressions.isEmpty`